music/src/piano_genie/model.ts

```diff
@@ -0,0 +1,295 @@
+/**
+ * Core implementation for Piano Genie model.
+ *
+ * @license
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Imports
+ */
+import * as tf from '@tensorflow/tfjs-core';
+
+/**
+ * Constants.
+ */
+const DATA_TIME_QUANTIZE_RATE = 31.25;
+const DATA_MAX_DISCRETE_TIMES = 32;
+const RNN_NLAYERS = 2;
+const RNN_NUNITS = 128;
+const NBUTTONS = 8;
+const NPIANOKEYS = 88;
+
+/**
+ * A type for keeping track of LSTM state activations.
+ *
+ * @param c The c parameters of the LSTM.
```

Comment:
Can we add words about what "c" and "h" means? At least in the actual documentation, if not in the actual property values